src/plugins/vis_types/vega/public/vega_view/vega_base_view.js

```diff
@@ -44,6 +45,52 @@ for (const funcName of Object.keys(vegaFunctions)) {
 
 const bypassToken = Symbol();
 
+function normalizeDate(date) {
+  if (typeof date === 'number' && !isNaN(date)) {
+    return date;
+  } else if (date instanceof Date) {
+    return date;
+  } else {
+    return String(normalizeObject(date));
```

Comment:
I see the following comment below: 

```
NaN (will be converted to null)
- undefined (key will be removed)
```
Whatever `normalizeObject` returns, the value will be converted to a string representation of 'null' and 'undefined'